src/calendar-app/workerUtils/worker/CalendarWorkerImpl.ts

```diff
@@ -0,0 +1,230 @@
+					err.columnNumber = colno
+					// @ts-ignore
+					err.fileName = source
+					this.sendError(err)
+				}
+
+				return true
+			}
+		}
+	}
+
+	get exposedInterface(): DelayedImpls<CommonWorkerInterface> {
+		return {
+			async loginFacade() {
+				return locator.login
+			},
+
+			async customerFacade() {
+				return locator.customer()
+			},
+
+			async giftCardFacade() {
+				return locator.giftCards()
+			},
+
+			async groupManagementFacade() {
+				return locator.groupManagement()
+			},
+
+			async configFacade() {
+				return locator.configFacade()
+			},
+
+			async calendarFacade() {
+				return locator.calendar()
+			},
+
+			async mailFacade() {
+				return locator.mail()
+			},
+
+			async shareFacade() {
+				return locator.share()
+			},
+
+			async cacheManagementFacade() {
+				return locator.cacheManagement()
+			},
+
+			async counterFacade() {
+				return locator.counters()
+			},
+
+			async bookingFacade() {
+				return locator.booking()
+			},
+
+			async mailAddressFacade() {
+				return locator.mailAddress()
+			},
+
+			async blobAccessTokenFacade() {
+				return locator.blobAccessToken
+			},
+
+			async blobFacade() {
+				return locator.blob()
+			},
+
+			async userManagementFacade() {
+				return locator.userManagement()
+			},
+
+			async recoverCodeFacade() {
+				return locator.recoverCode()
+			},
+
+			async restInterface() {
+				return locator.cache
+			},
+
+			async serviceExecutor() {
+				return locator.serviceExecutor
+			},
+
+			async cryptoFacade() {
+				return locator.crypto
+			},
+
+			async cacheStorage() {
+				return locator.cacheStorage
+			},
+
+			async sqlCipherFacade() {
+				return locator.sqlCipherFacade
+			},
+
+			async random() {
+				return {
+					async generateRandomNumber(nbrOfBytes: number) {
+						return random.generateRandomNumber(nbrOfBytes)
+					},
+				}
+			},
+
+			async eventBus() {
+				return locator.eventBusClient
+			},
+
+			async entropyFacade() {
+				return locator.entropyFacade
+			},
+
+			async workerFacade() {
+				return locator.workerFacade
+			},
+
+			async contactFacade() {
+				return locator.contactFacade()
+			},
+		}
+	}
+
+	queueCommands(exposedWorker: DelayedImpls<CommonWorkerInterface>): Commands<WorkerRequestType> {
+		return {
+			setup: async (message) => {
+				console.error("WorkerImpl: setup was called after bootstrap! message: ", message)
+			},
+			testEcho: (message) =>
+				Promise.resolve({
+					msg: ">>> " + message.args[0].msg,
+				}),
+			testError: (message) => {
+				const errorTypes = {
+					ProgrammingError,
+					CryptoError,
+					NotAuthenticatedError,
+				}
+				// @ts-ignore
+				let ErrorType = errorTypes[message.args[0].errorType]
+				return Promise.reject(new ErrorType(`wtf: ${message.args[0].errorType}`))
+			},
+			reset: (message: WorkerRequest) => {
+				return resetLocator()
+			},
+			restRequest: (message: WorkerRequest) => {
+				// This horror is to add auth headers to the admin client
```

Comment:
I like this comment! But if it is for the admin client we should be able to remove this function from the calendar app, right?